uefi-test-runner/src/boot/memory.rs

```diff
@@ -91,7 +92,8 @@ fn memory_map(bt: &BootServices) {
 
     let (_key, mut desc_iter) = bt
         .memory_map(&mut buffer)
-        .expect("Failed to retrieve UEFI memory map");
+        .expect("Failed to retrieve UEFI memory map")
+        .expect("Warnings encountered while retrieving memory map");
```

Comment:
Same here. I know it's theoretically possible to get an unspecified warning when calling any function, but if the spec doesn't make it clear what sort of warnings could arise, it would be better to just return an `Error`.

I mean, even if the application checks for a warning, if it doesn't know what the firmware-specific error code is, it won't be able to *safely* take meaningful action.

Reply:
That would be a possible interpretation indeed. However, EFI does reserve space for non-standard warning codes, so I'm not sure if we should trust one sentence in the middle of the spec on this, considering that they got a full paragraph of Blt pixel format wrong :wink: